sema/subyt/sinks.py

```
@@ -74,17 +75,19 @@ def add(
-        self._file_path = file_path
+        assert_writable(path_name, force_output)
+        self._file_path: Path = Path(path_name)
         self._force_output = force_output
-        if Path(file_path).exists():
-            self.mtimes = {file_path: os.stat(file_path).st_mtime}
+        if self._file_path.exists():
+            self.mtimes = {
+                str(self._file_path): self._file_path.stats().st_mtime
+            }
 
     def __repr__(self):
```

Ruff (0.8.2)

Missing return type annotation for special method `__repr__`

Add return type annotation: `str`

(ANN204)